src/components/popover-container/popover-container.spec.tsx

```diff
@@ -785,6 +802,73 @@ describe("open state when click event triggered", () => {
     expect(onCloseFn).not.toHaveBeenCalled();
   });
 
+  it("should close the container and call onClose when controlled and escape key is pressed", () => {
+    const onCloseFn = jest.fn();
+    const MockWrapper = () => {
+      const [open, setOpen] = React.useState(true);
+
+      return (
+        <PopoverContainer
+          title="PopoverContainerSettings"
+          open={open}
+          onClose={(e) => {
+            setOpen(false);
+            onCloseFn(e);
+          }}
+        />
+      );
+    };
+    const wrapper = mount(<MockWrapper />);
+
+    expect(wrapper.update().find(PopoverContainer).prop("open")).toBe(true);
```

Comment:
**comment**: do we need this assertion? You've designed the mock component to start off with `open={true}` so I'm not sure we need to assert on that - especially as the rest of the test doesn't check that it's changed to `false` (which I agree it doesn't need to), instead it checks that the `onCloseFn` has been called.

Reply:
I can have that removed. I only had this in as a few other tests in this spec file also have this assertion, so I went with the pattern that was already in the file.